aten/src/TH/THStorage.cpp

```diff
@@ -21,7 +23,7 @@ void THStorage_free(THStorage *storage) {
     if(--storage->refcount == 0)
     {
       if(storage->flag & TH_STORAGE_FREEMEM) {
-        storage->allocator->free(storage->allocatorContext, storage->data_ptr);
+        static_cast<THAllocator*>(storage->allocatorVoidPtr)->free(storage->allocatorContext, storage->data_ptr);
```

Comment:
I'm curious: why not use a union or other type?  There are only two types here, unlike e.g. scalar types, where a union would be more trouble.

Reply:
No particular reason. However, this change is getting promptly killed by #9089 so let's not fix it.

Reply:
sounds good.